WordPress/gutenberg
# Column Block: enable global template_lock inheritance

Issues to be fixed:
- (Main) #37912
- #17262

Related PRs: 
- (Should be able to closed) #17301
- #23329

## What?
This PR fixes a problem that template lock doesn't work correctly in `core/column` block.

As per the above issue and PR, this issue has been recognized for a long time, and I believe the discussion was centered on "**whether to let column blocks inherit template_lock**".

However, compared to the past, template locks and block locks have been enhanced.
I think it is reasonable now to allow column block to inherit `templateLock` in order to unify the behavior of the lock function.

## Why?
I confirmed that only  `core/column` block has the default value of `templateLock` set to false as attirbute.
https://github.com/WordPress/gutenberg/blob/5cca6b6b6dc2b91b2b599ec7533dc9a47239a706/packages/block-library/src/column/edit.js#L31

If I understand correctly, `templateLock` as attribute is `undefined` if `template_lock` is enabled throughout the editor.
Therefore, I think that the default value of `templateLock` for `core/column` block was `false` and applied to innnerBlocks.

In particular, a critical bug that is caused by template_lock not being inherited such as #37912 should be fixed.

## How?
Simply removed default values.

## Testing Instructions

- Insert a column block and one paragraph block on the post editor.
- Activate `template_lock` with the following code:

```php
function custom_registered_post_type_post( $post_type, $post_type_object ) {
  $post_type_object->template = array(
    array( 'core/paragraph' )
  );
  $post_type_object->template_lock = 'insert';
}
add_action( 'registered_post_type_post', 'custom_registered_post_type_post', 10, 2 );
```

- Reload the browser and confirm that the paragraph block in the column block is locked.
- Confirm that you cannot insert a new block within the column block.

Please see [this comment](https://github.com/WordPress/gutenberg/issues/37912#issuecomment-1194121020) for more information on the defects this PR is trying to fix.
